core/src/main/java/feign/Logger.java

```diff
@@ -89,7 +90,9 @@ protected Response logAndRebufferResponse(String configKey, Level logLevel, Resp
       }
 
       int bodyLength = 0;
-      if (response.body() != null) {
+      if (response.body() != null && !(status == 204 || status == 205)) {
```

Comment:
According to the rules below, it seems erroneous if there a body when status like this is sent? what http client are you using? could you add a MockWebServer test to show this? Why don't we fix the client instead?